support/oidc-discovery-provider/main.go

```diff
@@ -39,17 +39,24 @@ func run(configPath string) error {
 	}
 	defer log.Close()
 
-	if config.Domain != "" {
-		log.Warn("The `domain` configurable is deprecated and will be removed in a future release; use `domains` instead.")
-	}
-
 	source, err := newSource(log, config)
 	if err != nil {
 		return err
 	}
 	defer source.Close()
 
-	var handler http.Handler = NewHandler(config.Domains, source, config.AllowInsecureScheme, config.Domain == "")
+	domainPolicy, err := DomainAllowlist(config.Domains...)
+	if err != nil {
+		return err
+	}
+	if config.Domain != "" {
+		log.Warn("The domain configurable is deprecated and will be removed in a future release; use domains instead.")
+		// Maintain backwards compatibility and allow requests over all domains
+		// if configured with the deprecated `domain` configurable.
+		domainPolicy = AllowAnyDomain()
```

Comment:
Just want to make sure my understanding is correct here ... previously, we:
* Used `domain` value for forming URLs in the documents we serve here
* Used `domain` value for allow listing of ACME requests
* Did _not_ enforce that `domain` value is set as host header in HTTP handler

and now, we do the same except the last point is actually enforced. Is that right?

Reply:
Yup, that sums it up.